applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -483,6 +501,48 @@ void DistanceModificationProcess::SetDiscontinuousDistanceToSplitFlag()
     }
 }
 
+void DistanceModificationProcess::CheckAndStoreVariablesList(const std::vector<std::string>& rVariableStringArray) 
+{
+    const auto& r_node = *mrModelPart.NodesBegin(); 
+    if (rVariableStringArray.size()>0){
+        for (std::size_t i_variable=0; i_variable < rVariableStringArray.size(); i_variable++){
+            if (KratosComponents<Variable<double>>::Has(rVariableStringArray[i_variable])) {
+                const auto& r_double_var  = KratosComponents<Variable<double>>::Get(rVariableStringArray[i_variable]);
+                KRATOS_CHECK_DOF_IN_NODE(r_double_var, r_node);
+                KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(r_double_var, r_node)
+
+                mDoubleVariablesList.push_back(&r_double_var);               
+            }
+            else if (KratosComponents<ComponentType>::Has(rVariableStringArray[i_variable])){
+                const auto& r_component_var  = KratosComponents<ComponentType>::Get(rVariableStringArray[i_variable]);
+                KRATOS_CHECK_DOF_IN_NODE(r_component_var, r_node);
+                KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(r_component_var, r_node)
+               
+                mComponentVariablesList.push_back(&r_component_var);
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(rVariableStringArray[i_variable])){
```

Comment:
Here I check if its a vector variable and then I get each component with Kratos components. Do you like this approach?

Reply:
Look much better! Thanks @marcnunezc !